### PR TITLE
Refine build configurations on EC2 agents

### PR DESCRIPTION
**Pull request by blindpirate**
Fixes https://github.com/gradle/gradle-private/issues/2951

Fixed https://github.com/gradle/gradle-private/issues/2943

### Context

This PR adds an `EC2` tag to build scan.

This PR moves `org.gradle.internal.plugins.portal.url.override` property from build parameters to init script so we can disable this property on EC2 agents.

### CODE REVIEW

File: gradle/shared-with-buildSrc/mirrors.settings.gradle.kts [hunk 1]
```diff
@@ -53,7 +57,16 @@ fun withMirrors(handler: RepositoryHandler) {
 
 fun normalizeUrl(url: String): String {
     val result = url.replace("https://", "http://")
-    return if (result.endsWith("/")) result else result + "/"
+    return if (result.endsWith("/")) result else "$result/"
+}
+
+if (System.getProperty(PLUGIN_PORTAL_OVERRIDE_URL_PROPERTY) == null && !isEc2Agent() && !isMacAgent()) {
```
Comment (blindpirate) [inline gradle/shared-with-buildSrc/mirrors.settings.gradle.kts]: Previously we can set this property to `https://plugins.gradle.org/m2` to quickly disable the mirror settings if the mirror repo is down. But with init script how can do that quickly? Maybe we need another CI parameter to control this behavior?

File: gradle/shared-with-buildSrc/mirrors.settings.gradle.kts [hunk 2]
```diff
@@ -34,7 +35,12 @@ val mirrorUrls: Map<String, String> =
         name to url
     } ?: emptyMap()
 
-fun isEc2Agent() = java.net.InetAddress.getLocalHost().getHostName().startsWith("ip-")
+
+fun isEc2Agent() = java.net.InetAddress.getLocalHost().hostName.startsWith("ip-")
```
Comment (wolfs) [inline gradle/shared-with-buildSrc/mirrors.settings.gradle.kts]: Can you use the `HostnameLookup` service instead?

Comment (blindpirate) [reply gradle/shared-with-buildSrc/mirrors.settings.gradle.kts]: Two questions: 

- What's the advantage of `HostnameLookup`?
- Is it more convenient to be used in init script?

Comment (wolfs) [reply gradle/shared-with-buildSrc/mirrors.settings.gradle.kts]: > * What's the advantage of `HostnameLookup`?

It is faster.

> * Is it more convenient to be used in init script?

How easy is it to obtain a service in an init script?

**Pull request merged**